File: app/src/main/java/com/practice/nycschools/BaseActivity.java
package com.practice.nycschools;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class BaseActivity extends AppCompatActivity {

    String base = "https://data.cityofnewyork.us/";
    static Retrofit retrofit;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        if (retrofit == null)
            retrofit = new Retrofit.Builder()
                    .baseUrl(base)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

    }

    public Retrofit getRetrofit() {
        return retrofit;
    }
}

File: app/src/main/java/com/practice/nycschools/BaseActivity.kt
package com.practice.nycschools

import androidx.appcompat.app.AppCompatActivity
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

open class BaseActivity : AppCompatActivity() {

    companion object {
        private var base = "https://data.cityofnewyork.us/"
        var retrofit: Retrofit = Retrofit.Builder()
            .baseUrl(base)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }
}